tests/Signals_perftest2.m

```diff
@@ -0,0 +1,161 @@
+          post(testCase.B, 2);
+        % listener ('onValue') & 'map' ops
+        case {'onValue', 'map'}
+          f = @plus;
+          c = feval(Ops, testCase.A, @(x) f(1,x));
+          while (testCase.keepMeasuring)
+            post(testCase.A, 1);
+          end
+        % 'scan'
+        case {'scan'}
+          f = @plus;
+          c = feval(Ops, testCase.A, f, testCase.B);
+          post(testCase.B, 0);
+          while (testCase.keepMeasuring)
+            post(testCase.A, 1);
+          end
+        % 'subscriptable'
+        case {'subscriptable'}
+          s = struct('A', 1);
+          c = feval(Ops, testCase.A);
+          c_A = c.A;
+          while (testCase.keepMeasuring)
+            post(testCase.A, s);
+          end
+      end
+      
+      % test op
+      switch func2str(Ops)
+        case {'nop', 'post'}
+          while (testCase.keepMeasuring)
+            for i = 1:testCase.Reps
+              feval(Ops, testCase.A, 1);
+            end
+          end
+        case {'gt', 'ge', 'lt', 'le', 'eq', 'plus', 'minus', 'times',...
+            'rdivide', 'onValue', 'map', 'scan'}
+          while (testCase.keepMeasuring)
+            for i = 1:testCase.Reps
+              post(testCase.A, 1); % this post will trigger the update of signal 'c', defined above
+            end
+          end
+        case {'subscriptable'}
+          while (testCase.keepMeasuring)
+            for i = 1:testCase.Reps
+              post(testCase.A, s); % this post will trigger the update of signal 'c_A', defined above
+            end
+          end
+      end
+      
+    end
+    
+    function test_networkOps(testCase, OnlyBasicOps, NumNodes)
+      % benchmarks operations on a signals network upon posting to a signal
+      % which updates all other signals in the network
+      
+      % create an empty array that will grow
+      vals = [];
+      
+      % set-up network to test
+      otherSigs = cell(NumNodes,1);
+      if OnlyBasicOps % pick operation for signal at random amongst 'testCase.BasicOps'
+        for i = 1:NumNodes
+          op = randsample(testCase.BasicOps, 1);
+          op = op{1};
+          otherSigs{i} = feval(op, testCase.A, testCase.B);
+        end
+      else % pick operation for signal at random amongst 'testCase.Ops'
+        for i = 1:NumNodes
+          op = randsample(testCase.Ops, 1);
+          op = op{1};
+          switch func2str(op)
+            % for ops that do not create a new signal upon posting 1 to
+            % 'testCase.A', change op to '@gt'
+            case {'nop', 'post', 'subscriptable', 'onValue'}
+              op = @gt;
+              otherSigs{i} = feval(op, testCase.A, 1);
+            case {'gt', 'ge', 'lt', 'le', 'eq', 'plus', 'minus', 'times', 'rdivide'}
+              otherSigs{i} = feval(op, testCase.A, testCase.B);
+            case {'map'}
+              otherSigs{i} = feval(op, testCase.A, @(x) plus(1,x));
```

Comment:
If you're interested in testing the performance on map, I suggest using identity so that you are actually just measuring the function call time.  Other than adding scan, what does this do that isn't tested in the original code?

Reply:
ok. yeah some other small things that differed were adding tests for nop and assignment operations, and "depth" and "width" testing (which we talked about over skype) on networks of different sizes. i'll just combine the two classes into one 'perftest' class like you suggested

Reply:
Great, I suggest making a new separate test for methods that don't directly map to an op code.